pkg/supply/services.go

```diff
@@ -76,30 +74,72 @@ func LoadServiceCredentials(log *libbuildpack.Logger, serviceName string) (json.
 			}
 		}
 	}
+	var instanceID string
 	for _, amsSvc := range svcs[serviceName] {
+		instanceID = amsSvc.InstanceID
 		rawAmsCreds = append(rawAmsCreds, amsSvc.Credentials)
 	}
 	if len(rawAmsCreds) != 1 {
-		return json.RawMessage{}, fmt.Errorf("expect only one service (type %s or user-provided) but got %d", serviceName, len(rawAmsCreds))
+		return json.RawMessage{}, "", fmt.Errorf("expect only one service (type %s or user-provided) but got %d", serviceName, len(rawAmsCreds))
+	}
+	return rawAmsCreds[0], instanceID, nil
+}
+
+func LoadIASClientCert(log *libbuildpack.Logger) (cert []byte, key []byte, err error) {
+	iasCredsRaw, _, err := LoadServiceCredentials(log, "identity")
+	if err != nil {
+		return cert, key, err
 	}
-	return rawAmsCreds[0], nil
+	var iasCreds IASCredentials
+	err = json.Unmarshal(iasCredsRaw, &iasCreds)
+	if err != nil {
+		return cert, key, err
+	}
+	if iasCreds.Certificate == "" || iasCreds.Key == "" { // TODO: Provide option for {"credential_type":"X509_PROVIDED"}
+		return cert, key, fmt.Errorf("identity service binding does not contain client certificate. Please use binding parameter {\"credential_type\":\"X509_GENERATED\"}")
```

Comment:
Since we are making x509 certs required: How about using the validator for this check here?

Reply:
Thought about it too, but left it as is for the more verbose error message including the hint to crendential_type.